Internal: Fix implementation for the restore functionality in the styles command [EDS-354]

ronros-elementor opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

ronros-elementor committed:
added a new command to notify about any updates done to the style

ronros-elementor committed:
removed unused args

ronros-elementor committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
removed redundant jsdoc type

ronros-elementor committed:
Merge remote-tracking branch 'ronros-elementor/feature/EDS-354' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

Omerisra6 commented on modules/atomic-widgets/assets/js/editor/commands/styles.js:
```diff
@@ -43,6 +43,8 @@ export class Styles extends $e.modules.editor.document.CommandHistoryDebounceBas
 
 		container.model.set( 'styles', isRedo ? changes.new.styles : changes.old.styles );
+
+		$e.run( 'document/atomic-widgets/styles-updated', { container } );
```
Why does this command work with the history but the 'styles' command doesn't?

I think both commands will appear in the history...

Nevoss replied:
Seems like `settings` command uses in restore the same command:
<img width="809" alt="image" src="https://github.com/user-attachments/assets/8ebf3b14-057d-4044-ad2d-8084cbd6d5d4">

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
fixed relevant implementations in addToHistory + restore + apply to make history undo/redo work with the same command (for consistency purposes with the editor-style rendering package)

+ extracted the variant matching based on meta to a designated utils file

+ reverted the new empty command to notify of style changes

ronros-elementor committed:
Merge remote-tracking branch 'ronros-elementor/feature/EDS-354' into feature/EDS-354

ronros-elementor committed:
removed redundant use of initialStyleDefID (if started with no styleDefId - oldStyle would also not have it, so no need for this undefined conditional check really...

ronros-elementor committed:
removed redundant storage of settings

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

Nevoss commented on modules/atomic-widgets/assets/js/editor/commands/styles.js:
```diff
@@ -1,3 +1,5 @@
+import { getVariantByMeta } from '../utils/get-variants';
```
Make sure to remove the StyleDef and Variant if no need

ronros-elementor replied:
Done - added upon apply, checking first if no values other than `undefined` - if none it will remove the variant.

Then it takes the current style from the model again (as we're working with a structuralClone object), checks for the length of variants and if its 0 - will call for the internal command to delete the style

ronros-elementor committed:
taken care of the deletion of empty variants and empty style definitions

ronros-elementor commented on modules/atomic-widgets/assets/js/editor/commands/styles.js:
```diff
@@ -110,13 +125,18 @@ export class Styles extends $e.modules.editor.document.CommandHistoryDebounceBas
+				styleDefID,
+				bind,
+			} );
+			// Throw new Error( 'Style Def not found' );
```
Done

ronros-elementor commented on modules/atomic-widgets/assets/js/editor/commands/styles.js:
```diff
@@ -88,19 +107,15 @@ export class Styles extends $e.modules.editor.document.CommandHistoryDebounceBas
-
 	apply( args ) {
-		const { container, bind, meta, props } = args;
+		let { container } = args;
```
Done

ronros-elementor committed:
Merge remote-tracking branch 'ronros-elementor/feature/EDS-354' into feature/EDS-354

ronros-elementor committed:
updated tests

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

elementorbot committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
Merge branch 'main' into feature/EDS-354

ronros-elementor committed:
Merge branch 'main' into feature/EDS-354
